src/Catty/DataModel/VariablesContainer/UserDataContainer.m

```diff
@@ -174,41 +122,33 @@ - (UserList*)findUserListNamed:(NSString*)name inArray:(NSArray*)userLists
     return list;
 }
 
-- (void)removeObjectUserVariableNamed:(NSString*)name inArray:(NSMutableArray*)userVariables forSpriteObject:(SpriteObject*)sprite
+- (BOOL)removeUserVariableWithName:(NSString*)name
 {
-    pthread_mutex_lock(&variablesLock);
-    for (int i = 0; i < [userVariables count]; ++i) {
-        UserVariable *var = [userVariables objectAtIndex:i];
-        if ([var.name isEqualToString:name]) {
-            [userVariables removeObjectAtIndex:i];
-            [self.objectVariableList setObject:userVariables forKey:sprite];
-            break;
-        }
+    UserVariable *variable = [self getUserVariableWithName:name];
+    if (variable) {
+        [self removeProjectUserVariableNamed:name];
+        return YES;
     }
-    pthread_mutex_unlock(&variablesLock);
+    return NO;
 }
 
-- (void)removeObjectUserListNamed:(NSString*)name inArray:(NSMutableArray*)userLists forSpriteObject:(SpriteObject*)sprite
+- (BOOL)removeUserListWithName:(NSString*)name
 {
-    pthread_mutex_lock(&variablesLock);
-    for (int i = 0; i < [userLists count]; ++i) {
-        UserVariable *list = [userLists objectAtIndex:i];
-        if ([list.name isEqualToString:name]) {
-            [userLists removeObjectAtIndex:i];
-            [self.objectListOfLists setObject:userLists forKey:sprite];
-            break;
-        }
+    UserList *list = [self getUserListWithName:name];
+    if (list) {
+        [self removeProjectUserListNamed:name];
+        return YES;
     }
-    pthread_mutex_unlock(&variablesLock);
+    return NO;
 }
 
 - (void)removeProjectUserVariableNamed:(NSString*)name
```

Comment:
Please remove this deprecated method and remove the variable directly at `removeUserVariableWithName`.